modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java

```diff
@@ -876,23 +876,94 @@ public void testBindPublishedObjectDefinitions() throws Exception {
 
 		// Bind two published object definitions
 
+		ObjectDefinition objectDefinitionB =
+			_addAndPublishCustomObjectDefinition("B");
+		ObjectDefinition objectDefinitionBB =
+			_addAndPublishCustomObjectDefinition("BB");
+
 		_testBindObjectDefinitions(
-			_addAndPublishCustomObjectDefinition("A"),
-			_addAndPublishCustomObjectDefinition("AA"),
+			objectDefinitionB, objectDefinitionBB,
 			(objectDefinition1, objectDefinition2) ->
 				TreeTestUtil.assertObjectDefinitionTree(
 					LinkedHashMapBuilder.put(
-						"A", new String[] {"AA"}
+						"B", new String[] {"BB"}
 					).put(
-						"AA", new String[0]
+						"BB", new String[0]
 					).build(),
 					_treeFactory.createObjectDefinitionTree(
 						objectDefinition1.getObjectDefinitionId(),
 						_objectDefinitionLocalService::getObjectDefinition),
 					_objectDefinitionLocalService));
 
+		// Must have the same scope
+
+		ObjectDefinition siteObjectDefinition =
+			ObjectDefinitionTestUtil.publishObjectDefinition(
+				Collections.singletonList(
+					new TextObjectFieldBuilder(
+					).labelMap(
+						LocalizedMapUtil.getLocalizedMap(
+							RandomTestUtil.randomString())
+					).name(
+						"a" + RandomTestUtil.randomString()
+					).build()),
+				ObjectDefinitionConstants.SCOPE_SITE);
+
+		AssertUtils.assertFailure(
+			ObjectRelationshipEdgeException.class,
+			"Object definitions in a root context must have the same scope",
+			() -> _bindObjectDefinitions(
+				objectDefinitionB.getObjectDefinitionId(),
+				siteObjectDefinition.getObjectDefinitionId()));
+
+		// Must not form a cycle
+
+		ObjectDefinition objectDefinitionBBB =
+			_addAndPublishCustomObjectDefinition("BBB");
+
+		_bindObjectDefinitions(
+			objectDefinitionBB.getObjectDefinitionId(),
+			objectDefinitionBBB.getObjectDefinitionId());
+
+		AssertUtils.assertFailure(
+			ObjectRelationshipEdgeException.class,
+			"Object definitions in a root context must not form a cycle",
+			() -> _bindObjectDefinitions(
+				objectDefinitionBBB.getObjectDefinitionId(),
+				objectDefinitionB.getObjectDefinitionId()));
+
+		// Must not have more than one parent
```

Comment:
// Unable to bind the object definitions when the child object definition is bound to another object definition